### ISSUE TITLE
Inconsistent behaviour of onInvalidPaste

### ISSUE THREAD

**Issue by JamesKhoury** [autoNumeric/autoNumeric#658]
### Current behavior
An error is thrown when pasting an invalid value that is outside the min and max values when selecting all the text in the input.

### Expected behavior
That the `onInvalidPaste` option would always work consistently whether or not the user has selected the text.

### Steps to reproduce the problem
1. Tested on autoNumeric 4.1.0
2. In the browser `Chrome` version `77`
3. On the `Windows` operating system
4. Then instantiate the autoNumeric object with the following options:
```js
new AutoNumeric(domElement, {
  maximumValue: "99",
  minimumValue: "-99",
  onInvalidPaste: 'clamp'
});
```
5. Select inside the input. Do not highlight the whole value.
6. paste some invalid value (e.g. 11aa22)
7. See the result is as expected.
8. highlight the whole value in the input.
9. Paste the invalid value.
10. See error in the console.

### What we've tried
The `overrideMinMaxLimits: 'ignore'` option but then the input doesn't honor the limits at all. It would be better to be able to have the input 'clamp' the value on paste and not raise an error but still not allow the input to be set outside the min/max values by the user.

### Link to live example
 [jsFiddle](https://jsfiddle.net/r18eaov4/2/)

**Comment by AlexandreBonneau** [comment]
Indeed, since you are using `onInvalidPaste: 'clamp'`, selecting all the content of the field should not change the result, and clamp the value to the maximum allowed.

**Comment by kferenc617** [comment]
I've added a PR that solves this issue too, hopefully: PR #827
